### PR TITLE
Add useBearerSchemaForAuthorization to extauth

### PR DESCRIPTION
**Pull request by inFocus7**
# Description

Please include a summary of the changes.

This bug fixes ... \ This new feature can be used to ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by inFocus7** [comment]
/kick flake
```
Step #10 - "test": Summarizing 1 Failure:
Step #10 - "test": 
Step #10 - "test": [Fail] Access Log in memory Access Logs File [It] can create string access logs 
Step #10 - "test": /workspace/gloo/test/e2e/access_log_test.go:275
```

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 1]
```diff
@@ -384,6 +384,8 @@ message HeaderConfiguration {
     string id_token_header = 1;
     // If set, the access token will be forward upstream using this header name.
     string access_token_header = 2;
+    // If true, adds the "Bearer" prefix to the upstream access token header value
+    bool use_bearer_schema_for_authorization = 3;
```
Comment (nfuden) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:388]: I think for future proofing we want this to be a nullable bool
```suggestion
        google.protobuf.BoolValue use_bearer_schema_for_authorization = 3;
```

Comment (inFocus7) [reply projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto]: Done, thanks! 👍🏼

**Review by nfuden — APPROVED** [review]

File: projects/gloo/pkg/api/v1/enterprise/options/extauth/v1/extauth.pb.go [hunk 2]
```diff
@@ -1280,6 +1280,8 @@ type HeaderConfiguration struct {
 	IdTokenHeader string `protobuf:"bytes,1,opt,name=id_token_header,json=idTokenHeader,proto3" json:"id_token_header,omitempty"`
 	// If set, the access token will be forward upstream using this header name.
 	AccessTokenHeader string `protobuf:"bytes,2,opt,name=access_token_header,json=accessTokenHeader,proto3" json:"access_token_header,omitempty"`
+	// If true, adds the "Bearer" prefix to the upstream access token header value
```
Comment (Rachael-Graham) [inline projects/gloo/pkg/api/v1/enterprise/options/extauth/v1/extauth.pb.go:1283]: might need a period here? cant tell as it's rendered with one already [here](https://github.com/solo-io/gloo/pull/6576/files#diff-91cf81779d8c6085b561bb6652f224bf3675da302f30d567092cef4fd2d3157dR612)

**Review by Rachael-Graham — COMMENTED** [review]
docs - 1 question otherwise LGTM

**Review by nfuden — APPROVED** [review]

**Pull request merged**